openshift_metrics/utils.py

```diff
@@ -0,0 +1,142 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+#
+
+import datetime
+import json
+import requests
+import time
+
+import openshift as oc
+
+
+def query_metric(openshift_url, token, metric, report_date, retry=3):
+    attempt = 0
+    data = None
+    headers = {'Authorization': "Bearer %s" % token}
+    day_url_vars = "start=%sT00:00:00Z&end=%sT23:59:59Z" % (report_date, report_date)
+    print("Retrieving metric: %s" % metric)
+    while attempt < retry:
+        url = "%s/api/v1/query_range?query=%s&%s&step=60s" % (openshift_url, metric, day_url_vars)
+        r = requests.get(url, headers=headers, verify=False)
```

Comment:
Please don't disable ssl verification by default. Our production clusters should all be using valid ssl certificates. There should be a command line option to either disable ssl validation or to use a non-default certificate authority bundle (`verify` can be either a boolean or string, in which case it points to a ca bundle).

Reply:
Whoooops, good catch! Fixed.